project/PY-Chang/README.md

```diff
@@ -0,0 +1,59 @@
+# YAMM
+Yet Another Matrix Multiplier!
+
+## Basic Information
+> Github repository: https://github.com/PY-Chang/YAMM.git
+
+This is a matrix multiplier based on C++ and Python hybrid system. This project implements the classic Strassen algorithm with the help of parallel computing to achieve better performance on modern computer.
+
+## Problem to Solve and Prospective Users
+Matrix multiplication is one of the most important linear algebra operation in the modern computing world. For decades, many researches have been invested to improve the efficiency of matrix multiplication. 
+
+In 1969, Volker Strassen developed an algorithm, namely, Strassen algorithm which redeuces the time complicity of the matrix multiplication down to around O(n<sup>2.8074</sup>) in the first time. 
+
+
+Although in recent years researchers have developed faster algorithms than Strassen's, those algorithms have been viewed as galactic algorithms which means non-practical to present-day computers.
+
+Therefore in this project, we will implement the Strassen algorithm based on C++ and provide an easy-to-use python API to the user who is not familiar with the algorithm and parallel computing.
```

Comment:
Merely implementing the Strassen algorithm will be too small a scope for a term project.  It usually constitutes one homework assignment in a numerical-analysis or linear-algebra course.  And you will not be able to come up with an API that is more useful than LAPACK.  There will be little to say in your presenetation.

It is a good direction to do something with linear algebra, though.  I suggest you to cover more  work in linear algebra.

Reply:
@yungyuc What if I implement part of BLAS single precision routines and wrap them into python? The purpose is to let users use the same function prototype to call the API in the python as in c. (Although I know there has been CBLAS already...)

Reply:
If you plan to reimplement BLAS, you need to justify with the value of your work.  It does not need to be fancy, but you can reference some BLAS (re)implementations that are done over the years.